articles/language/file-structure.md

```diff
@@ -385,3 +385,32 @@ function DotProduct(a : Double[], b : Double[]) : Double {
     return accum;
 }
 ```
+
+
+## Internal Declarations
+
+User-defined types, operations, and functions can also be declared as *internal*.
+This means that they can only be accessed from within the Q# project that they are declared in.
+When a project is used as a reference, all of its *public* (non-internal) declarations are made available, but trying to use an internal declaration from another project will produce an error.
+Internal declarations are useful for writing modular code that can be reused by other parts of your project, but still be changed later without breaking other projects that might depend on it.
+
+> [!WARNING]
+> Internal user-defined types cannot be used in the signatures of public callables or in the underlying types of public user-defined types.
+> They can only be used in signatures or underlying types if the corresponding callable or user-defined type is also internal.
```

Comment:
I'd suggest adding a quick example here.
```suggestion
> They can only be used in signatures or underlying types if the corresponding callable or user-defined type is also internal.
>
> For example, if a UDT `InternalOptions` is defined with the `internal` keyword, then the following declarations will result in errors:
> ```Q#
> function DefaultInternalOptions() : InternalOptions { ... } // error: can't use InternalOptions as an output type
> newtype ExtendedOptions = (Internal : InternalOptions); // error: can't use InternalOptions as an item
> ```
```

Reply:
It looks like this PR was merged before I could commit your suggestions. I'll make another PR for them.

Reply:
Sorry about that. I can revert it.

Reply:
Committed in #765.